### ISSUE TITLE
Anime

### ISSUE THREAD

**Issue by hcgiub001** [Tikipeter/tikipeter.github.io#84]
I use to watch lots of anime, I always thought coco scrapers with addons like Fenlight are bad, I use Otaku/Haru
I saw this thread in reddit
https://www.reddit.com/r/Addons4Kodi/comments/1ela4bt/does_fenfen_lite_or_umbrella_provides_latest/

and played around with both Otaku and Fenlight. Fenlight pretty much gets sources for everything. maybe long airing animes it might struggle with but other than that it's really complete. It has so many positives (and some negatives) compared to Otaku.

to give you an example it's much faster. it has access to Extras with recommendations, trailers, cast etc
I can't even see a section in Otaku for recommended based on what you are watching.

so got me thinking. Fenlight already has this amazing platform but it's not specifically for anime so I can't filter by genre or airing this week or best anime of all time.

So I asked Bing Copilot which API is best to develop on for anime. 

You have
![Screenshot 2024-08-11 191902](https://github.com/user-attachments/assets/54e45982-d5ed-4d5e-a389-07ef7f70c0ba)

Myanimelist is more extensive but requires Auth 2.0 authentication2. but you have Jikan.

So I was thinking, in homepage, have an "Anime" section
when users click that you can use API from a site you think will be the quickest

just set up sections like 

-Airing anime
-movies
-shows
-trending
-popular
-completed
-top 100 anime
-genres
(using API's to get this data) 

basically what a normal addon has but for anime. We are not trying to make an extensive anime addon, just one that has the basic categories, why not, you already have links to it.

anime is very popular and seems like people prefer the speed and features of Fenlight.
no rush but would this interest you to take this on and integrate anime more into Fenlight?

![Screenshot 2024-08-11 192757](https://github.com/user-attachments/assets/761c44cd-6d8d-4e1f-b0cc-9710254bd75f)

again thanks for hearing me out.

**Comment by hcgiub001** [comment]
so I looked at anime and used the first "In Trakt List" which had 500 anime
https://trakt.tv/users/brendenmorris/lists/animes?sort=released,asc

I searched for first 200 series, I loaded up and played first episode of each season in each anime, sometimes I clicked more than 1 episode in that season and If first episode worked the rest did.

I made a list of each episode that didn't work and then I went into Otaku to check if Otaku could load them

here is my results

shin ikki tousen                         1
no estab-life: Great Escape              2
nobody the greatest demon it has         3
ss level 99: I may be hidden it has      4
Adam's sweet agony doesn't show up       0
4 cut hero doesn't come up               0
monster but wild doesn't come up         0
my life as inukai-san's dog              0
shin ikki 3rd episode works              5           
Estab-life: Great Escape doesn't have    0
Nobody the greatest demon it has         6
realm of legends works                   7
Tokyo 24th ward works                    8
spare me is not showing up in otaku but season 1 works in fenlight!!!!!!!!!!
ancient girls fame not working in otaku  0
build-divide works                       9
taisho otome                             10
vampire dies in no time season 2 working 11
D-cide traumerei working                 12
peach boy riverside working              13
full dive working                        14
don't call us a junk not working         0
hero return not working                  0
assault lilly working                    15
the idhun not working                    0
peter grill not showing up in otaku      0
demon spirit not working                 0
science fell in love working season 2 but no season 1  0

if it didn't work, it counts as 0, Spare Me, Great Lord! 2021 has 2 season, season 1 works in Fenlight but not season 2, and in Otaku it won't even return a search

so we have 15 out of 200 that didn't work, that's 92.5% that works, most of the ones that didn't work are older than a few years so basically everything new works,

this can't be a coincidence. something or someone has put effort for it to work, right?

92.5% is great for something that just works without any effort and we have access to searching other content using cast which isn't something Otaku can do.

Any chance you can look at anime API and see if you can make a section for anime in Fenlight and use feed that data into TMDB to give us a link we work with?

we just need normal lists like popular, trending , genre currently airing etc, all this from the API.
I think over time that 92.5% will get higher too.
peace

**Comment by hcgiub001** [comment]
so on this thread 
https://www.reddit.com/r/Addons4Kodi/comments/1epspuz/comment/lhosm3g/?context=3
kodifitzwell (POV dev) has actually got this working and better than I thought

this is his comment 
"POV/Dradis/Infinity have had a small section of Anime from the main menu for a while. it uses the Simkl api to grab some lists, but the meta still comes from tmdb. it's not as fast as the movies/tvshows list because its an extra api call and its not paginated yet. anything without a tmdb/imdb gets dropped as the watched status is dependent upon that.

also read the changelog because there is a scraper option not enabled by default."

so we can use Simkl api, I checked and it's good, though maybe even going with the anime specific API might be better?
he also added 
nyaa and nyaaio (pack capable)
![Screenshot (287)](https://github.com/user-attachments/assets/688dd25c-4887-480c-be8a-b0de0642f1e2)


and this pretty much gives all anime.
he only missed out on 3/200 anime but even found an anime that otaku didn't

not sure how adding scarpers work or where he's getting them from.

anyway, proof of concept works. You can have 98% of all anime and 100% of all recent anime inside Fenlight, I personally think it will be the best anime addon, it won't have 100% but it'll have 98% plus the integration of cast etc.

hope you consider this. even just using simkl seems great. kodifitzwell did take FEN as his base, you can use his ideas on simkl as yours?

**Comment by hcgiub001** [comment]
I tried adding whatever POV did to get more anime sources and came across this thread
https://www.reddit.com/r/Addons4Kodi/comments/1cjmiu3/is_it_possible_to_add_these_scrapers_to/

> Here, use this one instead: https://files.catbox.moe/eg0sds.py

> The previous one removed name filtering only from Nyaa. This one removes it for all Torrentio results. I just tested it with that 
> show and it gets a result now. You may occasionally get incorrect results now when Torrentio has mapped wrong (though I have 
> never run across this myself except for a remake with the same title once). But that's better than getting no results at all.

> Make sure you also clear Fen Light's cache.

I did as the person instructed and now I can play all anime that POV could, even thought he's using Nyaaio scraper and this is torrentio 
this scraper even found the brazillian show that the OP asked in that thread 
https://www.themoviedb.org/tv/239465-a-mao-do-eurico
and it wasn't found in POV, since POV isn't able to make use of that torrentio I think?

not sure how this helps going forward. I want a solution where it's easy for everyone and no need to do anything more. Fenlight is able to play 95% of anime I think now, good enough to have it's own mini anime section?

**Comment by TwilightMercy** [comment]
> I tried adding whatever POV did to get more anime sources and came across this thread
> 
> https://www.reddit.com/r/Addons4Kodi/comments/1cjmiu3/is_it_possible_to_add_these_scrapers_to/
> 

This manual torrentio.py change is no longer relevant since latest CocoScrapers versions has “bypass filter title” setting for Torrentio/Elfhosted sources.

**Comment by hcgiub001** [comment]
> > I tried adding whatever POV did to get more anime sources and came across this thread
> > https://www.reddit.com/r/Addons4Kodi/comments/1cjmiu3/is_it_possible_to_add_these_scrapers_to/
> 
> This manual torrentio.py change is no longer relevant since latest CocoScrapers versions has “bypass filter title” setting for Torrentio/Elfhosted sources, just turn it on.

that's awesome! makes it easier to do the changes for everyone, thankyou! are you the person who wrote it?

**Comment by Tikipeter** [comment]
Anime is a rabbit hole I am not really eager to go down. Fen Light's entire reliance on TMDb for metadata makes it a nightmare waiting to happen. I've looked at the API's listed, and all of them fall short of what I would want provided. Unlike kodifitzwell, I am unwilling to make an extra call to TMDb for every item in every list just to get basic ID's for the show.

I've made Fen Light's menu system as fluid as possible for situations like this. There are likely very good Trakt lists out there that service the Anime community pretty well with "Popular", "New", "Top Rated", "Top Viewed" etc lists ready to go.

You could easily make an "Anime" shortcut folder, add those Trakt lists to it, and then add that folder to the root menu or the TV Show menu and achieve a nice little Anime submenu.

If TMDb would open a new genre for anime, instead of including it in animated, then I could make useful lists using the TMDb API, but alas that is not the case.

**Comment by hcgiub001** [comment]
damn I had no idea, mdblist has basically list for stuff like trending, new for anime , and basically for everything! this is awesome!
now thinking how best to use this data and lists!
